lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -254,14 +266,49 @@ static Value createConstantOp(FIRRTLType opType, APInt value,
   return Value();
 }
 
-static Type getHandshakeBundleDataType(BundleType bundle) {
-  if (auto dataType = bundle.getElementType("data")) {
-    auto intType = dataType.cast<firrtl::IntType>();
-    return IntegerType::get(bundle.getContext(), intType.getWidthOrSentinel(),
+/// Creates a Value that has an assigned zero value. For bundles, this
+/// corresponds to assigning zero to each element recursively.
+static Value createZeroDataConst(FIRRTLType dataType, Location insertLoc,
+                                 OpBuilder &builder) {
+  if (dataType.isa<IntType>())
```

Comment:
The body of this function could be rewritten as ```return llvm::TypeSwitch<Type, Value>(dataType).Case... ``` although I wouldn't say it's mandatory due to "only" checking two cases - up to you.